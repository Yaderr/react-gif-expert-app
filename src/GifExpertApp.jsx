import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'Noticias', 'Tractor' ])

    const addNewCategory = (newCategory) => {
        if(categories.includes(newCategory)) return
        setCategories([newCategory, ...categories])
    }
    
    return (
        <>
            <h1>Gif Expert App</h1>
            <AddCategory onNewCategory={ addNewCategory } />
            {
                categories.map(category => (
                    <GifGrid key={ category } category={ category } />
                ))
            }            
        </>
    )
}