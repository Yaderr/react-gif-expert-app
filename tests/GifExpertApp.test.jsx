import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('GifExpertApp component tests', () => {
    
    test('debe hacer math con el snapshot', () => {

        const { container } = render(<GifExpertApp />)
        expect(container).toMatchSnapshot()
    })

    test('se deben mostrar las categorias iniciales', () => {
        const defaultCategories = [ 'Noticias', 'Tractor' ]
        
        render(<GifExpertApp />)
        
        expect(screen.getByText(defaultCategories[0]))
        expect(screen.getByText(defaultCategories[1]))
    })

    

    test('debe mostrar la categoria aÃ±adidad desde el form', () => {
        
        const category = 'NewCat'
        render(<GifExpertApp />)

        const input = screen.getByRole('searchbox')
        const form = screen.getByRole('form')
        
        fireEvent.input(input, {target: { value: category }})
        fireEvent.submit(form)

        expect(screen.getByText(category)).toBeTruthy()
    })
})