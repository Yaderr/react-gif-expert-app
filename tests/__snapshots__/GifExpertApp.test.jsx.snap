// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp component tests debe hacer math con el snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <div
    class="addCategory-container"
  >
    <form
      aria-label="form"
    >
      <input
        name=""
        placeholder="Busca todos lo gif"
        type="search"
        value=""
      />
    </form>
  </div>
  <div
    class="content"
  >
    <h2>
      Noticias
    </h2>
    <h2>
      Loading...
    </h2>
    <div
      class="gif-grid"
    />
  </div>
  <div
    class="content"
  >
    <h2>
      Tractor
    </h2>
    <h2>
      Loading...
    </h2>
    <div
      class="gif-grid"
    />
  </div>
</div>
`;
